src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -349,11 +349,16 @@ void MeshForwarder::RemoveDataResponseMessages(void)
 void MeshForwarder::SendMesh(Message &aMessage, Mac::TxFrame &aFrame)
 {
     uint16_t fcf;
+#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+    bool timeSync = aMessage.IsTimeSync();
```

Comment:
how about make Message::IsTimeSync() always return false when OPENTHREAD_CONFIG_TIME_SYNC_ENABLE is not 1.